src/deploy/campaign/environments.ts

```diff
@@ -117,8 +119,56 @@ export const getConfig = (
     rootPriceConfig: priceConfig,
     zeroVaultAddress: process.env.ZERO_VAULT_ADDRESS ? process.env.ZERO_VAULT_ADDRESS : zeroVault.address,
     mockMeowToken: process.env.MOCK_MEOW_TOKEN ? !!process.env.MOCK_MEOW_TOKEN : true,
-    stakingTokenAddress: process.env.STAKING_TOKEN_ADDRESS ? process.env.STAKING_TOKEN_ADDRESS : MEOW_TOKEN
+    stakingTokenAddress: process.env.STAKING_TOKEN_ADDRESS ? process.env.STAKING_TOKEN_ADDRESS : MEOW_TOKEN,
   };
 
+  // Will throw an error based on any invalid setup, given the `ENV_LEVEL` set
+  validate(config);
+
   return config;
 };
+
+// For testing the behaviour when we manipulate, we have an optional "env" string param
+export const validate = (config : IDeployCampaignConfig, env ?: string, mongoUri ?: string) => {
+  let envLevel;
+
+  if (env) {
+    envLevel = env;
+  } else {
+    envLevel = process.env.ENV_LEVEL;
+  }
+
+  if (envLevel === "dev") return; // No validation needed for dev
+
+  if (!mongoUri) mongoUri = process.env.MONGO_URI ? process.env.MONGO_URI : DEFAULT_MONGO_URI;
+
+  // Mainnet or testnet
+  if (envLevel === "prod" || envLevel === "test") {
+    requires(!config.mockMeowToken, "Cannot mock MEOW token in production");
+    requires(config.stakingTokenAddress === MEOW_TOKEN, "Must use MEOW token in production");
```

Comment:
use this here - https://github.com/zer0-os/zNS/blob/feat/deploy-campaign/src/deploy/missions/contracts/meow-token/mainnet-data.ts#L3